### DISCUSSION TITLE
W jaki sposob wyglada struktura aplikacji?

### DISCUSSION THREAD

**Discussion by wvffle** [pb-iot/backend#4] [Q&A]
@pb-iot/backend, czy mozecie mi powiedziec jak wyglada struktrura aplikacji? @Pyame wiem, ze juz o tym troche rozmawialismy przy obiedzie, ale chcialbym miec w 1 miejscu podsumowanie dokladne jak dokladnie to wyglada. Pomoze to ustalic jakie konkretnie widoki potrzebujemy

**Comment by Pyame** [comment]
# Struktura 
## User
- imie
- nazwisko
- email
- haslo
- rola (fk)
- aktywny

## Rola + uprawnienia jako enumy (do ustalenia czy nie wystarcza nam is_superuser itp i tylko do nich przypisywac uprawnienia)
- nazwa (obserwator/opiekun, pracownik, wlasciciel)
- lista uprawnien

## Urządzenie
- nazwa
- rodzaj (np. tablet, natryski)
- typ (funkcyjne, sensor)
- status (enumy)

## Lokacja
- nazwa
- dl geograficzna(new)
- szer geograficzna  (new)

## Szklarnia
- nazwa
- typ uprawy (new)
- lokacja (fk)
- lista urzadzen (fk)
- lista uprawnionych uzytkownikow (fk)
- wlasciciel (fk)
- default srodowisko (fk) (new)
- lista srodowisk (fk)  (new)

## Srodowisko
- data (new)
- temperatura
- wilgotność powietrza
- poziom światła
- PAR
- poziom CO2
- poziom wody w podłożu
- poziom zasolenia (EC) w podłożu
- temperatura podłoża
- waga podłoża i roślin
- mikro zmienność łodygi

## Autentykacja:
- JWT

## Funkcjonalności:
### Użytkownik:
- rejestracja
- logowanie
- przypisywanie roli (admin)
- dodawanie szklarni
- edycja/usuwanie konta

### Szklarnia:
- włączanie/wyłączanie urządzeń funkcyjnych
- zebranie pomiarów
- informowanie o awarii sprzętu
- dodawanie/edycja/usuwanie urządzeń
- manualne sterowanie sprzętem
- statystyki

**Comment by patrycjarecko** [comment]
@wvffle na razie taki jest zamysl, jestesmy otwarci na inne pomysly
